subsys/logging/log_core.c

```diff
@@ -62,11 +64,121 @@ static k_tid_t proc_tid;
 static u32_t dummy_timestamp(void);
 static timestamp_get_t timestamp_func = dummy_timestamp;
 
+
+bool log_is_strdup(const void *buf);
+
 static u32_t dummy_timestamp(void)
 {
 	return 0;
 }
 
+/**
+ * @brief Count number of string format specifiers (%s).
+ *
+ * Result is stored as the mask (argument n is n'th bit). Bit is set if %s was
+ * found.
+ *
+ * @note Algorithm does not take into account complex format specifiers as they
+ *	 hardly used in log messages and including them would significantly
+ *	 extended this function which is called on every log message is feature
+ *	 is enabled.
+ *
+ * @param str String.
+ * @param nargs Number of arguments in the string.
+ *
+ * @return Mask with %s format specifiers found.
+ */
+static u32_t count_s(const char *str, u32_t nargs)
+{
+	#define IS_LOWER(c)	(((c) >= 'a') && ((c) <= 'z'))
+	char curr;
+	bool arm = false;
+	u32_t arg = 0;
+	u32_t mask = 0;
+
+	__ASSERT_NO_MSG(nargs <= 8*sizeof(mask));
+
+	while ((curr = *str++) && arg < nargs) {
+		if (curr == '%') {
+			arm = !arm;
+		} else if (arm && IS_LOWER(curr)) {
```

Comment:
Have you tried isalpha()?

Reply:
that would include capital letters as well? Aren't capital letters used for modifiers?

Reply:
0x%08X

Reply:
right, changed to use `isalpha`